### PR TITLE
chore: remove travis and autopkgtest

### PR DESCRIPTION
**Pull request by mr-cal**
- [x] Have you followed the [guidelines for contributing](https://github.com/canonical/snapcraft/blob/main/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox run -m lint`?
- [x] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----

Remove unused code related to travis and autopkgtest.

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -98,8 +98,9 @@ tox run -e lint-codespell
 
 ## Evaluating pull requests
 
-Oftentimes all you want to do is see if a given pull request solves the issue you were having. To make this easier, the Travis CI setup for snapcraft _publishes_ the resulting snap that was built for x86-64 using `transfer.sh`.
-To download the snap, find the relevant CI job run for the PR under review and locate the "snap" stage, the URL to download from will be located at the end of logs for that job.
+Oftentimes all you want to do is see if a given pull request solves the issue you were having. To make this easier, a snap is published for ``amd64`` on a channel named ``latest/edge/pr-<PR number>`` where ``PR number`` is the number of the pull request.
+
+For feature branches, a snap is published for ``amd64`` on a channel named `latest/feature/<branch name>`.
```
Comment (lengau) [inline HACKING.md:103]: `latest/edge/feature/<branch name>`?

File: snapcraft/store/client.py [hunk 2]
```diff
@@ -34,8 +34,6 @@
 from ._legacy_account import LegacyUbuntuOne
 from .onprem_client import ON_PREM_ENDPOINTS, OnPremClient
 
-_TESTING_ENV_PREFIXES = ["TRAVIS", "AUTOPKGTEST_TMP"]
```
Comment (lengau) [inline snapcraft/store/client.py:37]: If you want to keep the user-agent change, you could detect the `CI` and/or [`PYTEST_CURRENT_TEST`](https://docs.pytest.org/en/latest/example/simple.html#pytest-current-test-environment-variable).

I'm not sure how valuable this is, but it is an option.

File: HACKING.md (hunk 1 above)
Comment (mr-cal) [reply HACKING.md]: Good catch, I corrected it and added an example.

**Review by lengau — APPROVED** [review]
Very nice!

**Review by tigarmo — APPROVED** [review]

File: snapcraft/store/client.py (hunk 2 above)
Comment (mr-cal) [reply snapcraft/store/client.py:37]: Since no one is needing it, I'll omit it for simplicity.

**Pull request merged**